ytxp-plutarch/src/Cardano/YTxP/Control/Yielding/Scripts.hs

```diff
@@ -1,73 +1,71 @@
 module Cardano.YTxP.Control.Yielding.Scripts (
-  compileYieldingMP,
-  compileYieldingValidator,
-  compileYieldingSV,
+  scripts,
 ) where
 
 import Cardano.YTxP.Control.Yielding.Helper (yieldingHelper)
-import Cardano.YTxP.SDK.SdkParameters (AuthorisedScriptsSTCS)
-import Data.Text (Text)
-import Numeric.Natural (Natural)
-import Plutarch (Config, compile)
-import Plutarch.Api.V2 (PScriptContext)
-import Plutarch.Script (Script)
+import Data.Map (fromList)
+import Data.Text (Text, unpack)
+import Plutarch (Config)
+import Plutarch.LedgerApi (PCurrencySymbol, PScriptContext)
+import Ply (TypedScriptEnvelope)
+import Ply.Plutarch.TypedWriter (TypedWriter, mkEnvelope)
+import ScriptExport.ScriptInfo (RawScriptExport (RawScriptExport))
 
 --------------------------------------------------------------------------------
--- Yielding Validator Script
+-- Raw Script Export
 
-compileYieldingValidator ::
-  Config ->
-  AuthorisedScriptsSTCS ->
-  Either
-    Text
-    Script
-compileYieldingValidator config ylstcs = do
-  let
-    yieldingValidator ::
-      forall (s :: S).
-      ( Term s (PData :--> PData :--> PScriptContext :--> POpaque)
-      )
-    -- Takes the @yieldingHelper@ and turn it into a 3 argument script
-    yieldingValidator = plam $ \_datum redeemer ctx ->
-      yieldingHelper ylstcs # redeemer # ctx
-
-  compile config yieldingValidator
+scripts :: Config -> RawScriptExport
+scripts conf =
+  RawScriptExport $
+    fromList
+      [ envelope "djed:yieldingValidator" yieldingValidator
+      , envelope "djed:yieldingMP" yieldingMP
+      , envelope "djed:yieldingSV" yieldingSV
+      ]
+  where
+    envelope ::
+      forall (pt :: S -> Type).
+      (TypedWriter pt) =>
+      Text ->
+      ClosedTerm pt ->
+      (Text, TypedScriptEnvelope)
+    envelope d t = (d, either (error . unpack) id $ mkEnvelope conf d t)
 
 --------------------------------------------------------------------------------
--- Yielding Minting Policy Script
-
-compileYieldingMP ::
-  Config ->
-  AuthorisedScriptsSTCS ->
-  Natural ->
-  Either
-    Text
-    Script
-compileYieldingMP config stcs nonce = do
-  let
-    yieldingMP ::
-      forall (s :: S).
-      ( Term s (PData :--> PScriptContext :--> POpaque)
-      )
-    yieldingMP = plet (pconstant $ toInteger nonce) (const $ yieldingHelper stcs)
+-- Plutarch level terms
 
-  compile config yieldingMP
+-- | Yielding Validator
+yieldingValidator ::
+  forall (s :: S).
+  Term
+    s
+    ( PCurrencySymbol :--> PData :--> PData :--> PScriptContext :--> POpaque
+    )
+yieldingValidator = plam $ \psymbol _datum redeemer ctx ->
+  yieldingHelper # psymbol # redeemer # ctx
```

Comment:
@albertodvp we should adapt the tests to use these instead of the yielding helper directly (although it doesn't really change much)